backend/src/main/java/com/skr/virtuallibrary/services/BookListService.java

```diff
@@ -0,0 +1,147 @@
+package com.skr.virtuallibrary.services;
+
+import com.skr.virtuallibrary.dto.BookListDto;
+import com.skr.virtuallibrary.entities.BookList;
+import com.skr.virtuallibrary.entities.User;
+import com.skr.virtuallibrary.entities.enums.Language;
+import com.skr.virtuallibrary.exceptions.AccessForbiddenException;
+import com.skr.virtuallibrary.exceptions.BookListAlreadyExistsException;
+import com.skr.virtuallibrary.exceptions.BookListNotFoundException;
+import com.skr.virtuallibrary.exceptions.BookNotFoundException;
+import com.skr.virtuallibrary.mapping.ModelMapper;
+import com.skr.virtuallibrary.repositories.BookListRepository;
+import com.skr.virtuallibrary.repositories.BookRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class BookListService {
+
+    private final BookListRepository bookListRepository;
+
+    private final UserService userService;
+
+    private final BookRepository bookRepository;
+
+    private final ModelMapper modelMapper;
+
+    private static final String BOOK_LIST_NOT_FOUND = "Book list not found wit id: ";
+
+    private static final String ACCESS_DENIED = "You don't have access to this book list.";
+
+    public List<BookListDto> getBookLists() {
+        User user = userService.getCurrentUser();
+        List<BookList> bookLists = bookListRepository.findAllByUserId(user.getId());
+        if (bookLists.isEmpty() || bookListRepository.findAllByNameAndUserId(toReadListName(user), user.getId()).isEmpty()) {
```

Comment:
Jeśli pierwszy warunek jest prawdziwy, to drugi też

Reply:
poprawione